app/models/screening.rb

```diff
@@ -23,19 +23,17 @@ def start_nil?
     start_time.nil?
   end
 
-  def used?
-    return if hall_available?
-
-    errors.add(:start_time, 'Hall is used for another screening')
+  def time_range
+    start_time..end_time
   end
 
-  def hall_available?
-    Screening
-      .where(hall_id:)
-      .where(end_time: start_time..end_time)
-      .or(Screening.where(hall_id:)
-      .where(start_time: start_time..end_time))
-      .where.not(id:)
-      .empty?
+  def used?
+    screenings = Screening.where(hall_id:).where.not(id:)
```

Comment:
it sound like a scope :D then you could use it in the `used?` method instead of assigning a variable